### PR TITLE
sadf: Allow epoch time for start and end time flags

### PR DESCRIPTION
**Pull request by NatePW**
Current sadf allows for start and end time flags to be given as hh:mm[:ss], but this can lead to unexpected results since the -s and -e flags expect the time in UTC, and new files are created at 00:00 local time.

For example, if local time zone has a -7 hour offset, and I wish to query for records starting after 17:00 local time, the start flag would have to be 24:00 (17:00 + 7: hour offset) or higher. That would understandable not allowed due to input validation, so it would have to roll over to the morning hours. Since there is no date specified with this format, sadf would return records from the beginning of the file.

In an effort to remove ambiguity of the desired date, this PR adds the ability to input a start and end time in UTC as number of seconds from the epoch.

### CODE REVIEW

File: sa_common.c [hunk 1]
```diff
@@ -714,6 +761,12 @@ int parse_timestamp(char *argv[], int *opt, struct tstamp *tse,
 				strncpy(timestamp, argv[(*opt)++], 8);
 				break;
 
+			case 10:
+				strncpy(timestamp, argv[(*opt)++], 10);
+				timestamp[10] = '\0';
+				return decode_epoch(timestamp, tse);
+				break;
```
Comment (portante) [inline sa_common.c:768]: Break not needed here.

**Review by portante — COMMENTED** [review]
Seems reasonable.

**Comment by stevekay** [comment]
I expect a few sysstat man page will need revising to reflect this new functionality added to `-s` and `-e`.

**Comment by stevekay** [comment]
`parse_timestamp` function header to be updated to include new epoch date functionality.

> Parse a timestamp entered on the command line (hh:mm[:ss]) and decode it.

**Comment by sysstat** [comment]
Hi,
Each record saved by sadc has an associated time stamp (HH:MM:SS). This time stamp is expressed in local time by default unless the environment variable `S_TIME_DEF_TIME` is set to "UTC", in which case it is saved in UTC. This environment variable is also used by sadc to determine when a file rotation shall happen (when the output file has been set to "-").
Anyway, the start and end times given as arguments with options `-s` and `-e` are not expressed in local time or UTC. They are merely "numbers" used in a comparison with the time stamp (expressed in local time or UTC) associated with each record.

**Comment by sysstat** [comment]
I have merged your PR but also updated your changes.

**Pull request closed without merging**